service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/FutureTaskResultToDomainObjectPromiseWrapper.java

```diff
@@ -108,9 +123,36 @@ public DomainObject get() throws PromiseBrokenException, InterruptedException {
         }
     }
 
+    private DomainObject get(ResponseMerger<DomainObject> merger) throws PromiseBrokenException, InterruptedException {
+        try {
+            ServiceResponse<DomainObject> response = new ServiceResponse<>();
+            for (Future<TaskResult> futureResult : futureList) {
+                TaskResult result = futureResult.get();
+                if (result == null) {
+                    throw new PromiseBrokenException("Task result is null");
+                }
+                triggerListeners();
```

Comment:
We can't use existing triggerListeners() which reads from single future object and not from futureList object. Add another method which handles triggering listeners with TaskResult from each iteration.